packages/starlight/utils/route-data.ts

```diff
@@ -66,9 +66,10 @@ function getToC({ entry, locale, headings }: PageProps) {
 	};
 }
 
-function getLastUpdated({ entry, id }: PageProps): Date | undefined {
+function getLastUpdated({ entry, id, isFallback }: PageProps): Date | undefined {
 	if (entry.data.lastUpdated ?? config.lastUpdated) {
-		const currentFilePath = fileURLToPath(new URL('src/content/docs/' + id, project.root));
+		const currentId = isFallback ? localizedId(id, config.defaultLocale.locale) : id;
+		const currentFilePath = fileURLToPath(new URL('src/content/docs/' + currentId, project.root));
```

Comment:
I actually think we can simplify it to this!

```suggestion
		const currentFilePath = fileURLToPath(new URL('src/content/docs/' + entry.id, project.root));
```

`PageProps` includes a root-level `id` localized for the current page, but `entry` also includes the original `id` for the content being used.